src/libraries/System.Resources.Extensions/src/System/Resources/Extensions/BinaryFormat/BinaryFormattedObjectExtensions.cs

```diff
@@ -0,0 +1,393 @@
+        {
+            hashtable = null;
+
+            // Note that hashtables with custom comparers and/or hash code providers will have that information before
+            // the value pair arrays.
+            if (format.RootRecord.RecordType != RecordType.SystemClassWithMembersAndTypes
+                || format.RootRecord is not ClassRecord classInfo
+                || !classInfo.IsTypeNameMatching(typeof(Hashtable))
+                || !classInfo.HasMember("Keys")
+                || !classInfo.HasMember("Values")
+                || format[2] is not ArrayRecord<object?> keysRecord
+                || format[3] is not ArrayRecord<object?> valuesRecord
+                || keysRecord.Length != valuesRecord.Length)
+            {
+                return false;
+            }
+
+            Hashtable temp = new((int)keysRecord.Length);
+            object?[] keys = keysRecord.ToArray(maxLength: ArrayRecordDeserializer.MaxArrayLength);
+            object?[] values = valuesRecord.ToArray(maxLength: ArrayRecordDeserializer.MaxArrayLength);
+            for (int i = 0; i < keys.Length; i++)
+            {
+                object? key = keys[i];
+                object? value = values[i];
+
+                if (key is null or SerializationRecord || value is SerializationRecord)
+                {
+                    return false;
+                }
+
+                temp[key] = value;
+            }
+
+            hashtable = temp;
+            return true;
+        }
+    }
+
+    /// <summary>
+    ///  Tries to get this object as a binary formatted <see cref="NotSupportedException"/>.
+    /// </summary>
+    public static bool TryGetNotSupportedException(
+        this BinaryFormattedObject format,
+        out object? exception)
+    {
+        return TryGet(Get, format, out exception);
+
+        static bool Get(BinaryFormattedObject format, [NotNullWhen(true)] out object? exception)
+        {
+            exception = null;
+
+            if (format.RootRecord is not ClassRecord classInfo
+                || classInfo.IsTypeNameMatching(typeof(NotSupportedException)))
+            {
+                return false;
+            }
+
+            exception = new NotSupportedException(classInfo.GetString("Message"));
+            return true;
+        }
+    }
+
+    /// <summary>
+    ///  Try to get a supported .NET type object (not WinForms).
+    /// </summary>
+    public static bool TryGetFrameworkObject(
+        this BinaryFormattedObject format,
+        [NotNullWhen(true)] out object? value)
+        => format.TryGetPrimitiveType(out value)
+            || format.TryGetPrimitiveList(out value)
+            || format.TryGetPrimitiveArray(out value)
+            || format.TryGetPrimitiveArrayList(out value)
+            || format.TryGetPrimitiveHashtable(out value)
+            || format.TryGetRectangleF(out value)
+            || format.TryGetPointF(out value)
```

Comment:
In https://github.com/dotnet/runtime/pull/102379/files/cabd585ef3ce00975cb5d3f9f795b6085d942776#diff-9b2940d51e511970ce444d9601af77f90696b76b1c906810e12806fda6a0b692 it was asked why this is needed, and the answer was that it isn't / most of this file could be deleted.

But (and I've checked that I'm not in a commit-limited view now) I still see this code here.

Reply:
> I still see this code here.

Great catch 👍